go-controller/pkg/node/iptables/iptables.go

```diff
@@ -17,12 +17,79 @@ type Rule struct {
 	Protocol iptables.Protocol
 }
 
+// AddRulesFiltered adds the given rules to iptables.
+// filter is a map[table][chain] of valid tables/chains to use for filtering rules to be added.
+func AddRulesFiltered(rules []Rule, append bool, filter map[string]map[string]bool) error {
+	addErrors := errors.New("")
+	var err error
+	var ipt util.IPTablesHelper
+	var exists bool
+
+	// stores valid table chains and whether they were already created or not
+	// key is ip protocol, table, chain
+	createdChains := map[iptables.Protocol]map[string]map[string]bool{
```

Comment:
We dont need to check if chains are present or not anymore, because with restore it will just recreate everything. I think maybe what you are alluding to is not making a bunch of recreate calls per chain and just doing a big recreate call for all the rules? That could be done, but if one of those chains has a broken rule in it then the entire restore may fail. I think the restore is pretty fast now so it doesn't buy us much performance.

Reply:
ack so I think perhaps your future commits are editing this commit because I see github made my comment outdated, so I only finished commit1, let me review your restore commit then it will make more sense.